### PR TITLE
Fixes free product cart rule not being added

### LINKED ISSUE #15751: 'Total available for each user' in cart rule not considered
**Describe the bug**
When you offer a product with price 0 as a free gift any record is inserted at table `ps_order_cart_rule`, so the customer can use it another time

**To Reproduce**

1. Set a product with price 0
image
1. Create a cart rule limited to 1 use per customer and offer this product as free gift:
image
image

1. Create an order. Cart rule is added correctly.
1. Create a second order. **Cart rule is added but it shouldn't**.

**Additional information**
PrestaShop version: 1.7.6.1

### PR DESCRIPTION
**Pull request by gennaris**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x 
| Description?      | Customers were able to reuse a cart rule intended for a one-time free gift because the entry in ps_order_cart_rule wasn't recorded when the cart total had a zero price. This PR introduces a check to ensure that cart rules of the “gift product” type are correctly accounted for, regardless of the cart total.
| Type?             | bug fix
| Category?         |  CO 
| BC breaks?        |  no
| Deprecations?     | no
| How to test?      | Check https://github.com/PrestaShop/PrestaShop/issues/15751 steps to reproduce before / after this patch
| UI Tests          | https://github.com/florine2623/testing_pr/actions/runs/12069290895 ✅ 
| Fixed issue or discussion?     | Fixes #15751 
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

**Review by kpodemski — APPROVED** [review]

**Review by Hlavtox — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @gennaris ,

Tested the standard case. Tested with odrer made from BO and FO.
Tested with the same customer and different customer. Tested with visitor and guest account.

Works as expected.
I can only have the voucher once.

It is QA ✅ 
Thanks!

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Comment by kpodemski** [comment]
thanks @gennaris 👌

**Pull request merged**
